service/src/main/java/bio/terra/pipelines/common/utils/FileUtils.java

```diff
@@ -105,4 +105,20 @@ public static String getStorageContainerUrlFromSasUrl(String sasUrl, UUID worksp
   public static String getFileNameFromFullPath(String fullPath) {
     return fullPath.substring(fullPath.lastIndexOf('/') + 1);
   }
+
+  /**
+   * Construct a path to a file given a base path and a file name. Adds a '/' between the base path
+   * and the file name if the base path does not end with a '/' or the file name does not begin with
+   * a '/'. Ensures there is only one '/' if both base and file paths contain a '/'.
+   */
+  public static String constructFilePath(String basePath, String fileName) {
+    String pathDelimiter = "/";
+    if (basePath.endsWith(pathDelimiter) && fileName.startsWith(pathDelimiter)) {
```

Comment:
i think we can use the stripTrailing and stripLeading functions in [stringutils](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html) to just get rid of all leading/trailing `/` then just add one ourselves

Reply:
oh that's nice! thanks

Reply:
those methods don't take parameters and only remove whitespace, but we can achieve the same thing using replace.